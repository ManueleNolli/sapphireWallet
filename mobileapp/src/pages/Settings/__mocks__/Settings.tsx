const SettingsMock = () => <div data-testid={'Settings'}>Settings</div>

export default SettingsMock
