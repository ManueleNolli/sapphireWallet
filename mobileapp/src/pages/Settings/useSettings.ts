import React, { useContext } from 'react'
import { ThemeContext } from '../../context/ThemeContext'
import { FirstAccessContext } from '../../context/FirstAccessContext'
import { Icon } from '@ui-kitten/components'
import { ImageProps } from 'react-native'

export default function useSettings() {
  const { theme, toggleTheme } = useContext(ThemeContext)
  const { toggleFirstAccess } = useContext(FirstAccessContext)
  const themeIconRef = React.useRef<Icon<Partial<ImageProps>>>()

  const toggleThemeWithAnimation = () => {
    if (themeIconRef.current) themeIconRef.current.startAnimation()
    toggleTheme()
  }

  return {
    theme,
    toggleFirstAccess,
    toggleThemeWithAnimation,
    themeIconRef,
  }
}
