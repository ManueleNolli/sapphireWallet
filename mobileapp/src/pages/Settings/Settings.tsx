import React, { useContext } from 'react'
import {
  Button,
  Divider,
  Icon,
  IconElement,
  Radio,
  RadioGroup,
  Text,
} from '@ui-kitten/components'
import SafeAreaView from '../../utils/SafeAreaView'
import { vh } from '../../Styles'
import useSettings from './useSettings'
import { NETWORKS } from '../../constants/Networks'
import {
  requestERC721TokenTransfer,
  requestETHBridgeCall,
} from '../../services/transactions'
import Toast from 'react-native-toast-message'
import { WalletContext } from '../../context/WalletContext'
import { getSigner } from '../../services/wallet'
import { BlockchainContext } from '../../context/BlockchainContext'

export default function Settings() {
  const {
    theme,
    resetLocalWallet,
    toggleThemeWithAnimation,
    themeIconRef,
    selectedIndex,
    onNetworkSelect,
  } = useSettings()

  const { getWalletContractAddress, getPrivateKey } = useContext(WalletContext)
  const { currentNetwork } = useContext(BlockchainContext)

  const ThemeIcon = (props: any): IconElement => (
    <Icon
      {...props}
      name={theme === 'light' ? 'moon' : 'sun'}
      animation="zoom"
      ref={themeIconRef}
    />
  )

  const DeleteIcon = (props: any): IconElement => (
    <Icon {...props} name="alert-triangle" />
  )

  const networkUppercase = (network: string) =>
    network.charAt(0).toUpperCase() + network.slice(1)

  const temp = async () => {
    try {
      await requestETHBridgeCall(
        getWalletContractAddress(),
        getWalletContractAddress(),
        0.02,
        await getSigner(
          await getPrivateKey('Sign transaction to send ETH'),
          currentNetwork
        ),
        currentNetwork
      )
    } catch (e: any) {
      console.log('catched erorr', e)
    }
  }

  return (
    <SafeAreaView style={{ paddingTop: 5 * vh }}>
      <Text category="h6">Network</Text>
      <Divider style={{ marginTop: 0.5 * vh, marginBottom: 2 * vh }} />
      <RadioGroup selectedIndex={selectedIndex} onChange={onNetworkSelect}>
        {Object.values(NETWORKS).map((network, index) => (
          <Radio key={index}>
            <Text category="s1">{networkUppercase(network)}</Text>
          </Radio>
        ))}
      </RadioGroup>

      <Text category="h6" style={{ marginTop: 10 * vh }}>
        Customisation
      </Text>
      <Divider style={{ marginTop: 0.5 * vh, marginBottom: 2 * vh }} />
      <Button
        style={{ width: 10 * vh, height: 10 * vh }}
        onPress={toggleThemeWithAnimation}
        accessoryLeft={ThemeIcon}
      />

      <Text category="h6" style={{ marginTop: 10 * vh }}>
        Dev settings
      </Text>
      <Divider style={{ marginTop: 0.5 * vh, marginBottom: 2 * vh }} />
      <Button
        appearance="outline"
        status="danger"
        accessoryLeft={DeleteIcon}
        onPress={resetLocalWallet}
      >
        Reset local wallet
      </Button>

      <Button status="danger" onPress={temp}>
        Temp
      </Button>
    </SafeAreaView>
  )
}
