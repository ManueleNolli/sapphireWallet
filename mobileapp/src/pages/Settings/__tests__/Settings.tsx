import renderWithTheme from '../../../TestHelper'
import Settings from '../Settings'
import useSettings from '../useSettings'

jest.mock('../useSettings', () => jest.fn())

describe('Settings', () => {
  beforeEach(() => {
    ;(useSettings as jest.Mock).mockReturnValue({
      theme: 'light',
      toggleFirstAccess: jest.fn(),
      toggleThemeWithAnimation: jest.fn(),
      themeIconRef: { current: { startAnimation: jest.fn() } },
    })
  })

  it('renders correctly', () => {
    const { toJSON } = renderWithTheme(<Settings />)
    expect(toJSON()).toMatchSnapshot()
  })
})
