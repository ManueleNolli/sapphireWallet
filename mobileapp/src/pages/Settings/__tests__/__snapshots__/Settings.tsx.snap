// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings renders correctly 1`] = `
<View
  appearance="default"
  level="1"
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
      },
      {
        "flex": 1,
      },
    ]
  }
>
  <RNCSafeAreaView
    edges={
      {
        "bottom": "additive",
        "left": "additive",
        "right": "additive",
        "top": "additive",
      }
    }
    style={
      [
        {
          "flex": 1,
          "marginHorizontal": 37.5,
          "marginVertical": 66.7,
        },
        undefined,
      ]
    }
  >
    <Text
      appearance="default"
      category="h1"
      status="basic"
      style={
        [
          {
            "color": "#222B45",
            "fontFamily": "System",
            "fontSize": 36,
            "fontWeight": "800",
          },
          {
            "marginBottom": 26.68,
          },
        ]
      }
    >
      Settings
    </Text>
    <Text
      appearance="default"
      category="h6"
      status="basic"
      style={
        [
          {
            "color": "#222B45",
            "fontFamily": "System",
            "fontSize": 18,
            "fontWeight": "800",
          },
          undefined,
        ]
      }
    >
      Customisation
    </Text>
    <View
      appearance="default"
      style={
        [
          {
            "backgroundColor": "#EDF1F7",
            "height": 1,
          },
          {
            "marginBottom": 26.68,
            "marginTop": 6.67,
          },
        ]
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#3366FF",
          "borderColor": "#3366FF",
          "borderRadius": 4,
          "borderWidth": 1,
          "flexDirection": "row",
          "height": 133.4,
          "justifyContent": "center",
          "minHeight": 40,
          "minWidth": 40,
          "opacity": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 12,
          "width": 133.4,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "transform": [
              {
                "scale": 1,
              },
            ],
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="20"
          bbWidth="20"
          fill="#FFFFFF"
          focusable={false}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "height": 20,
                "marginHorizontal": 10,
                "tintColor": "#FFFFFF",
                "width": 20,
              },
              {
                "flex": 0,
                "height": 20,
                "width": 20,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
        >
          <RNSVGGroup
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            propList={
              [
                "fill",
              ]
            }
          >
            <RNSVGGroup
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
            >
              <RNSVGGroup
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              >
                <RNSVGRect
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  height="24"
                  opacity={0}
                  width="24"
                  x="0"
                  y="0"
                />
                <RNSVGPath
                  d="M12.3 22h-.1a10.31 10.31 0 0 1-7.34-3.15 10.46 10.46 0 0 1-.26-14 10.13 10.13 0 0 1 4-2.74 1 1 0 0 1 1.06.22 1 1 0 0 1 .24 1 8.4 8.4 0 0 0 1.94 8.81 8.47 8.47 0 0 0 8.83 1.94 1 1 0 0 1 1.27 1.29A10.16 10.16 0 0 1 19.6 19a10.28 10.28 0 0 1-7.3 3z"
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                />
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
    <Text
      appearance="default"
      category="h6"
      status="basic"
      style={
        [
          {
            "color": "#222B45",
            "fontFamily": "System",
            "fontSize": 18,
            "fontWeight": "800",
          },
          {
            "marginTop": 133.4,
          },
        ]
      }
    >
      Dev settings
    </Text>
    <View
      appearance="default"
      style={
        [
          {
            "backgroundColor": "#EDF1F7",
            "height": 1,
          },
          {
            "marginBottom": 26.68,
            "marginTop": 6.67,
          },
        ]
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 61, 113, 0.08)",
          "borderColor": "#FF3D71",
          "borderRadius": 4,
          "borderWidth": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "minHeight": 40,
          "minWidth": 40,
          "opacity": 1,
          "paddingHorizontal": 10,
          "paddingVertical": 12,
        }
      }
    >
      <View
        collapsable={false}
        style={
          {
            "transform": [
              {
                "scale": 1,
              },
            ],
          }
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="20"
          bbWidth="20"
          fill="#FF3D71"
          focusable={false}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {
                "height": 20,
                "marginHorizontal": 10,
                "tintColor": "#FF3D71",
                "width": 20,
              },
              {
                "flex": 0,
                "height": 20,
                "width": 20,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
        >
          <RNSVGGroup
            fill={
              {
                "payload": 4294917489,
                "type": 0,
              }
            }
            propList={
              [
                "fill",
              ]
            }
          >
            <RNSVGGroup
              fill={
                {
                  "payload": 4278190080,
                  "type": 0,
                }
              }
            >
              <RNSVGGroup
                fill={
                  {
                    "payload": 4278190080,
                    "type": 0,
                  }
                }
              >
                <RNSVGRect
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                  height="24"
                  matrix={
                    [
                      6.123233995736766e-17,
                      1,
                      -1,
                      6.123233995736766e-17,
                      24,
                      0,
                    ]
                  }
                  opacity={0}
                  width="24"
                  x="0"
                  y="0"
                />
                <RNSVGPath
                  d="M22.56 16.3L14.89 3.58a3.43 3.43 0 0 0-5.78 0L1.44 16.3a3 3 0 0 0-.05 3A3.37 3.37 0 0 0 4.33 21h15.34a3.37 3.37 0 0 0 2.94-1.66 3 3 0 0 0-.05-3.04zM12 17a1 1 0 1 1 1-1 1 1 0 0 1-1 1zm1-4a1 1 0 0 1-2 0V9a1 1 0 0 1 2 0z"
                  fill={
                    {
                      "payload": 4278190080,
                      "type": 0,
                    }
                  }
                />
              </RNSVGGroup>
            </RNSVGGroup>
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <Text
        appearance="default"
        category="p1"
        status="basic"
        style={
          [
            {
              "color": "#222B45",
              "fontFamily": "System",
              "fontSize": 15,
              "fontWeight": "400",
            },
            {
              "color": "#FF3D71",
              "fontFamily": "System",
              "fontSize": 14,
              "fontWeight": "bold",
              "marginHorizontal": 10,
            },
          ]
        }
      >
        Reset local wallet
      </Text>
    </View>
  </RNCSafeAreaView>
</View>
`;
